onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -371,6 +408,31 @@ class ReduceAggregatorMax : public ReduceAggregator<T, TVAL> {
           }
         });
   }
+
+  static void FastReduceRKR(const Tensor& input, const std::vector<int64_t>& fast_shape,
+                            Tensor& output, concurrency::ThreadPool* tp) {
+    const T* data = input.Data<T>();
+    T* out = output.MutableData<T>();
+    int64_t d0 = fast_shape[0];
+    int64_t d2 = fast_shape[2];
+    int64_t inc = d2 * fast_shape[1];
+
+    concurrency::ThreadPool::TryParallelFor(
+        tp, fast_shape[1], ParallelReduceFastCost(fast_shape[1], fast_shape[0] * fast_shape[2], sizeof(T), 6),
+        [data, out, d0, d2, inc](ptrdiff_t begin, ptrdiff_t last) {
+          const T* p;
+          for (ptrdiff_t d = begin; d < last; ++d) {
+            p = data + d * d2;
+            out[d] = p[0];
+            for (int64_t i = 0; i < d0; ++i) {
+              for (int64_t k = 0; k < d2; ++k) {
+                out[d] = p[k] > out[d] ? p[k] : out[d];
+              }
```

Comment:
assuming this is faster than an `if` to conditionally update out[d]. would be good to capture that in a comment so it doesn't get changed in the future.

Reply:
I should check in this case. Here is what I did a while ago to count element. Compilers seem to prefer ``? :`` to optimize branching. But that may change depending the compiler version.

![image](https://user-images.githubusercontent.com/22452781/150329247-7181423b-9920-4742-94fa-69fbc95a3f7a.png)

Reply:
done